apps/meteor/tests/e2e/omnichannel/omnichannel-livechat-bot-routing.spec.ts

```diff
@@ -0,0 +1,162 @@
+
+	const visitorBot = createFakeVisitor();
+	const visitorHuman = createFakeVisitor();
+
+	test.beforeEach(async ({ browser, api }) => {
+		const context = await browser.newContext();
+		const page = await context.newPage();
+
+		poLiveChat = new OmnichannelLiveChat(page, api);
+		await poLiveChat.page.goto('/livechat');
+	});
+
+	test.beforeAll(async ({ api, browser }) => {
+		await api.post('/users.update', {
+			userId: 'user1',
+			data: {
+				roles: ['user', 'bot'],
+			},
+		});
+		await api.post('/livechat/users/agent', { username: 'user1' });
+		await api.post('/livechat/users/agent', { username: 'user2' });
+
+		await api.post('/settings/Livechat_Routing_Method', { value: 'Manual_Selection' });
+		await api.post('/settings/Livechat_assign_new_conversation_to_bot', { value: true });
+
+		const { page: omniPage } = await createAuxContext(browser, Users.user1, '/', true);
+		poHomeOmnichannel = new HomeOmnichannel(omniPage);
+
+		const { page: omniPageUser2 } = await createAuxContext(browser, Users.user2, '/', true);
+		poHomeOmnichannelUser2 = new HomeOmnichannel(omniPageUser2);
+	});
+
+	test.afterAll(async ({ api }) => {
+		await Promise.all([
+			api.post('/settings/Livechat_Routing_Method', { value: 'Auto_Selection' }),
+			api.post('/settings/Livechat_assign_new_conversation_to_bot', { value: false }),
+			api.delete('/livechat/users/agent/user1'),
+			api.delete('/livechat/users/agent/user2'),
+			api.post('/users.update', {
+				userId: 'user1',
+				data: {
+					roles: ['user'],
+				},
+			}),
+		]);
+
+		await poHomeOmnichannel.page.close();
+		await poHomeOmnichannelUser2.page.close();
+	});
+
+	test.afterEach(async () => {
+		await poLiveChat.closeChat();
+		await poLiveChat.page.close();
+	});
+
+	test('should prioritize bot agent over human agents when setting is enabled', async () => {
+		await test.step('visitor starts a chat', async () => {
+			await poLiveChat.openAnyLiveChatAndSendMessage({
+				liveChatUser: visitorBot,
+				message: 'Hello, I should be assigned to a bot agent',
+				isOffline: false,
+			});
+		});
+
+		await test.step('verify chat is assigned to bot', async () => {
+			const botQueuedChat = poHomeOmnichannel.sidenav.getQueuedChat(visitorBot.name);
+			await expect(botQueuedChat).toBeVisible();
+		});
+		await test.step('verify non-bot agent does not see the inquiry', async () => {
+			const nonBotQueuedChat = poHomeOmnichannelUser2.sidenav.getQueuedChat(visitorBot.name);
+			await expect(nonBotQueuedChat).toHaveCount(0);
+		});
+
+		await test.step('bot agent can see the chat', async () => {
+			await poHomeOmnichannel.sidenav.getQueuedChat(visitorBot.name).click();
+			await expect(poHomeOmnichannel.content.lastSystemMessageBody).toHaveText('joined the channel');
+		});
+	});
+
+	// Chat is not being assigned to any agent when "Assign New Conversations to Bot Agent" is enabled and bot is not available
+	test.fixme('should route to bot agent even when bot is offline', async () => {
```

Comment:
the test is wrong or you should use `test.fail` ?

Reply:
When I was creating the tests I found an issue related to it: [CORE-1065]( https://rocketchat.atlassian.net/browse/CORE-1065)
[discussion on channel](https://open.rocket.chat/group/RC-Core?msg=YDHBnejXj232XzkjM)
I added the "fixme" so once the behavior is fixed the tests should be working properly

> If the setting is enabled and a bot user/agent is found the assignment should occurs.

Reply:
ok but in theory `fixme` skips the test so if the fix is released or if the by magic the test start to pass no one will notice. the fails make sure that this test should fail because is a known bug. otherwise if was fixed the `fail` should be removed

Reply:
oh, got it, makes sense! changed!